src/cookieconsent.test.js

```diff
@@ -282,18 +282,22 @@ describe('onload', () => {
 
   test('enables the appropriate scripts', () => {
     const spy = jest.fn();
+    setConsentSetting('preferences', true);
```

Comment:
These tests are supposed to test the 'onload' behaviour, so we need to change the tests to reflect the new behaviour.
To answer your question, I don't think we should be calling setConsentSetting here. We should have other tests that test the setConsentSetting function.

Reply:
So it should be `expect(spy).toNotHaveBeenCalled()`, since we expect the enableScriptsByCategories function to have not been called.

Reply:
Will change now